include/grpcpp/ext/orca_service.h

```diff
@@ -48,38 +49,16 @@ class OrcaService : public Service {
     }
   };
 
-  explicit OrcaService(Options options);
-
-  // Sets or removes the CPU utilization value to be reported to clients.
-  void SetCpuUtilization(double cpu_utilization);
-  void DeleteCpuUtilization();
-
-  // Sets of removes the memory utilization value to be reported to clients.
-  void SetMemoryUtilization(double memory_utilization);
-  void DeleteMemoryUtilization();
-
-  // Sets of removes the QPS value to be reported to clients.
-  void SetQps(double qps);
-  void DeleteQps();
-
-  // Sets or removed named utilization values to be reported to clients.
-  void SetNamedUtilization(std::string name, double utilization);
-  void DeleteNamedUtilization(const std::string& name);
-  void SetAllNamedUtilization(std::map<std::string, double> named_utilization);
+  OrcaService(const ServerMetricRecorder& server_metric_recorder,
+              Options options);
 
  private:
   class Reactor;
 
-  Slice GetOrCreateSerializedResponse();
+  Slice CreateSerializedResponse();
 
+  const ServerMetricRecorder& server_metric_recorder_;
   const absl::Duration min_report_duration_;
-
-  grpc::internal::Mutex mu_;
-  double cpu_utilization_ ABSL_GUARDED_BY(&mu_) = -1;
-  double memory_utilization_ ABSL_GUARDED_BY(&mu_) = -1;
-  double qps_ ABSL_GUARDED_BY(&mu_) = -1;
-  std::map<std::string, double> named_utilization_ ABSL_GUARDED_BY(&mu_);
-  absl::optional<Slice> response_slice_ ABSL_GUARDED_BY(&mu_);
```

Comment:
This change loses the optimization that we currently have here to avoid reserializing the ORCA proto on each individual response.  I think we do want to try to retain that optimization, because a given `OrcaService` could have a very large number of clients, all of which will be getting the same data.

One possible option here would be to have `ServerMetricRecorder` track a timestamp for the last time any of its fields were updated, and have `OrcaService` cache the serialized proto along with the last time it grabbed data from `ServerMetricRecorder`.  That way, `OrcaService` could reserialize the proto only when the data has changed.  But we might need to use a mutex instead of separate atomics for each field in order to make that work properly.

Reply:
Updated, PTAL.

I introduced an atomic monotonically increasing counter. It guarantees that the caller can reuse the result if the sequence number is identical but not the inverse i.e. dup serialization with data race. I think this is fine while allowing us to keep atomic for root-level metrics but lmk.